### ISSUE TITLE
Potential optimizations

### ISSUE THREAD

**Issue by polyfractal** [pelias/pelias#5]
This looks like a very cool project!  I heard about it on Twitter and took a look...very impressive!  I just wanted to drop off a few potential optimizations

Feel free to ignore any/all of these, especially if you've already tried them out. :)
#### Heap Usage

It was mentioned on twitter (and the readme) that pelias uses a bunch of heap.  There are a few things you can do to reduce heap usage:
- If the data is static after import, you could disable bloom filters on each index.  The bloom filters are used to speed up indexing, but if the data is static, it just represents wasted heap space.  [Details about unloading are here](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/index-modules-codec.html#bloom-postings) (see the "tip").
- Similarly, if the data is fairly static, you can probably reduce your primary shard count.  Extra shards laying around will eat up heap space from Lucene overhead (term dictionaries, etc) and by reducing inverted index compression.  If you reduce the shard count to 40 or even 20 primary shards, you will save some memory.  This could potentially increase your query throughput, but may also increase the query latency (since queries will be mostly CPU bound by the geo_\* filters, decreasing primary shards decreases how many machines participate in each query)
#### Ingestion speed
- The readme says it takes three days ingest data?  Is that all Elasticsearch or also other components?  I would bump your bulk size to around 1000 docs.  The dataset is 66m docs and 300gb, so each doc is probably around 4kb.  Bumping the bulk size to 1000 docs will put you around 5-6mb per bulk, which is more optimal.
- Do you spin up multiple import processes/threads?  This can drastically reduce ingestion time
#### Query optimizations

The search looks to mostly be a demonstration, so you may not care about these, but a few easy wins:
- You may consider [enabling `lat_lon` for the geo_points](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/mapping-geo-point-type.html#_indexed_fields).  This indexes the lat/lon as individual fields, which enables the geo_\* filters to execute ranges on the inverted index instead of field data.  This is sometimes faster depending on the data involved.
- The ["closest" query](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/mapping-geo-point-type.html#_indexed_fields) can be restructured into a filtered query, which will potentially be faster.  Something like:

```
{
    "query": {
        "filtered": {
           "filter": {
               "and": {
                  "filters": [
                     {"term": {"location.type": "search.type"}},
                     {"geo_distance": {}}
                  ]
               }
           }
        }
    }
}
```

This does a few things.  First, it executes the Term as a filter instead of a query, which allows caching and should be faster.  Second, it executes the Term filter before the expensive geo because of the And compound filter.

Unclear if this would be faster, however, since your usage of the top-level filter will do a good job limiting the number of documents that the geo will see.  You might also try a hybrid approach like this:

```
{
    "query": {
        "filtered": {
           "filter": {
              "term": {"location.type": "search.type"}
           }
        }
    },
    "post_filter" : {
        "geo_distance: {}
    }
}
```

Basically identical to the existing query, except it uses a filter instead of a term query.
